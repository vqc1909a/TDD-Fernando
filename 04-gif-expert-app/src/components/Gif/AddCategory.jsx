import { useState } from "react"
import PropTypes from 'prop-types'


const DEFAULT_VALUE = "";
 
export const AddCategory = ({handleAddCategory}) => {
  
  const [inputValue, setInputValue] = useState(DEFAULT_VALUE);

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  }

  const handleSubmit = (e) => { 
    e.preventDefault(); 
    if(inputValue.trim()){
      handleAddCategory(inputValue);
      setInputValue(DEFAULT_VALUE);
    }
  } 
  return (
    <form onSubmit={handleSubmit}>
        <input type="text" placeholder="Buscar gifts" value={inputValue} onChange={handleInputChange}/>
    </form>
  )
}

AddCategory.propTypes = {
  handleAddCategory: PropTypes.func.isRequired
}