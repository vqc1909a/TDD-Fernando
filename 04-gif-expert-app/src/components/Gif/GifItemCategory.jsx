import { PropTypes } from "prop-types"

export const GifItemCategory = ({title, url}) => {
  return (
    <div className="card" aria-label="card">
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  )
}
GifItemCategory.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

/* Tarea

1. AÃ±adir PropTypes a GifItemCategory
  a. title obligatorio
  b. url obligatorio

2. Evaluar el snapshot

*/
