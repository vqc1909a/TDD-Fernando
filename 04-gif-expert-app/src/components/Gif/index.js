//This is a file of exports. It is used to export all the components from the folder Gif, This is also called "barrel" file
export * from "./AddCategory";
export * from "./GifGridCategory";
export * from "./GifItemCategory";

