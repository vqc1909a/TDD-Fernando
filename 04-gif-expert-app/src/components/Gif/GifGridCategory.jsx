import { PropTypes } from "prop-types"
import { GifItemCategory } from "../../components/Gif";
import { useFetchGifs } from "../../hooks/useFetchGifs";

export const GifGridCategory = ({category}) => {  
  const { gifs, isLoading }  = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>  
      {
        isLoading
        ?
        <p>Loading...</p>
        :
        <div className="card-grid" aria-label="card-grif">
          {gifs.map(image => <GifItemCategory key={image.id} {...image}></GifItemCategory>)}
        </div>
      }
    </>
  )
}

GifGridCategory.propTypes = {
  category: PropTypes.string.isRequired
}

