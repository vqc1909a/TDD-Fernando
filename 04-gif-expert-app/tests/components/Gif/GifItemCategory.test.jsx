import {render, screen} from "@testing-library/react";
import { GifItemCategory } from "../../../src/components/Gif/GifItemCategory"

describe('tests on component GifItemCategory', () => {

    const title = "Saitama";
    const url = "https://giphy.com/saitama.gif";

    test('component should make to match to the snapshot', () => {
        const {container} = render(<GifItemCategory title={title} url={url}></GifItemCategory>);
        expect(container).toMatchSnapshot();
    })

    test('should show the image and alt indicated correctly', () => { 
        render(<GifItemCategory title={title} url={url}></GifItemCategory>);

        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title);
    })

    test('should show the title in the component', () => { 
        render(<GifItemCategory title={title} url={url}></GifItemCategory>);

        expect(screen.getByText(title)).toBeTruthy();
        expect(screen.getByText(title).innerHTML).toContain(title);
    })
 })