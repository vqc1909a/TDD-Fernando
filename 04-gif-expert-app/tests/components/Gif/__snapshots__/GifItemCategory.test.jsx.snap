// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on component GifItemCategory component should make to match to the snapshot 1`] = `
<div>
  <div
    aria-label="card"
    class="card"
  >
    <img
      alt="Saitama"
      src="https://giphy.com/saitama.gif"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
